spec.bs

```diff
@@ -8159,43 +8321,154 @@ To <dfn>batch or fetch trusted bidding signals</dfn> given a [=trusted bidding s
+        new [=list=].
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        |putativeIgNames|.
+          « |igName| ».
       1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-        |putativeLengthLimit|.
+        |ig|'s [=interest group/max trusted bidding signals url length=].
   1. Otherwise:
-    1. [=Fetch the current outstanding trusted signals batch=] given |trustedBiddingSignalsBatcher|,
-      |signalsUrl|, |scriptOrigin|, |experimentGroupId|, |topLevelOrigin|, |slotSizeQueryParam|, and
-      |policyContainer|.
-    1. If |ig|'s [=interest group/trusted bidding signals keys=] is not null, set
-      |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-      [=list/clone=] of |ig|'s [=interest group/trusted bidding signals keys=].
-    1. Otherwise, set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-       new [=list=].
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        « |igName| ».
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-      |ig|'s [=interest group/max trusted bidding signals url length=].
+    1. Let |keyValueInterestGroups| be |trustedBiddingSignalsBatcher|'s [=trusted bidding signals
+      batcher/key value interest groups=].
+    1. Let |key| be [=tuple=] of (|ig|'s [=interest group/joining origin=], |ig|'s [=interest group/owner=]).
+    1. If |keyValueInterestGroups| [=map/contains=] |key|, [=set/append=] |ig| to
+      |keyValueInterestGroups|[|key|].
+    1. Otherwise, set |keyValueInterestGroups|[|key|] to a [=set=] « |ig| ».
+
+</div>
+
+<div algorithm>
+
+To <dfn>build trusted key value bidding signals request body</dfn> given a [=set=] |interestGroups|,
+an {{unsigned short}}-or-null |experimentGroupId|, an [=origin=] |topLevelOrigin|, a [=string=]
+|slotSizeQueryParam|, an [=origin=] |coordinator|, and an [=origin=] |owner|:
+
+1. Let |compressionGroups| be an empty [=list=], whose [=list/items=] are [=compression groups=].
+1. Let |compressionGroupMap| be an empty [=map=], where the [=map/keys=] are integers as
+  [=compression group/compression group id=], and the [=map/values=] are [=maps=] with integers as [=map/keys=]
+  for [=partition/id=], and [=partitions=] as their [=map/values=].
+1. Let |compressionIdMap| be an empty [=map=], whose [=map/keys=] are [=origins=] and [=map/values=] are integers.
+1. Let |interestGroupIdMap| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=tuples=]
+  of (interger, integer).
+1. Let |slotSizeParams| be the result of [=strictly splitting=] |slotSizeQueryParam| on U+003D (=).
+1. Let |nextCompressionGroupId| be 0.
+1. [=list/For each=] |group| of |interestGroups|:
+  1. Set |joiningOrigin| to |group|'s [=interest group/joining origin=].
+  1. Let |partitionId| be an integer.

```

Comment:
You don't need to declare it as integer. IIUC, you can directly set it to 0, and then at line 8399, you just set it to a different value if execution mode is not "`group-by-origin`"